cpp/scenario/gazebo/include/scenario/gazebo/Joint.h

```diff
@@ -0,0 +1,185 @@
+    bool createECMResources();
+
+    bool historyOfAppliedJointForcesEnabled() const;
+    bool enableHistoryOfAppliedJointForces( //
+        const bool enable,
+        const size_t maxHistorySize = 100);
+    std::vector<double> getHistoryOfAppliedJointForces() const;
+
+    // ==========
+    // Joint Core
+    // ==========
+
+    size_t dofs() const;
+    std::string name() const;
+    base::JointType type() const;
+
+    base::Limit positionLimit() const;
+
+    base::JointControlMode controlMode() const;
+    bool setControlMode(const base::JointControlMode mode);
+
+    double controllerPeriod() const;
+
+    double maxEffort() const;
+    bool setMaxEffort(const double maxEffort);
+
+    base::PID pid() const;
+    bool setPID(const base::PID& pid);
+
+    // ==================
+    // Single DOF methods
+    // ==================
+
+    double force(const size_t dof = 0) const;
+    double position(const size_t dof = 0) const;
+    double velocity(const size_t dof = 0) const;
+
+    bool setForce(const double force, const size_t dof = 0);
+    bool setPositionTarget(const double position, const size_t dof = 0);
+    bool setVelocityTarget(const double velocity, const size_t dof = 0);
+    bool setAccelerationTarget(const double acceleration, const size_t dof = 0);
+
+    double positionTarget(const size_t dof = 0) const;
+    double velocityTarget(const size_t dof = 0) const;
+    double accelerationTarget(const size_t dof = 0) const;
+
+    bool resetPosition(const double position, const size_t dof = 0);
+    bool resetVelocity(const double velocity, const size_t dof = 0);
+    bool reset(const double position, //
+               const double velocity,
+               const size_t dof = 0);
+
+    // =================
+    // Multi DOF methods
+    // =================
+
+    std::vector<double> jointForce() const;
+    std::vector<double> jointPosition() const;
+    std::vector<double> jointVelocity() const;
+
+    bool setJointForce(const std::vector<double>& force);
+    bool setJointPositionTarget(const std::vector<double>& position);
+    bool setJointVelocityTarget(const std::vector<double>& velocity);
+    bool setJointAccelerationTarget(const std::vector<double>& acceleration);
+
+    std::vector<double> jointPositionTarget() const;
+    std::vector<double> jointVelocityTarget() const;
+    std::vector<double> jointAccelerationTarget() const;
+
+    bool resetJointPosition(const std::vector<double>& position);
+    bool resetJointVelocity(const std::vector<double>& velocity);
+    bool resetJoint(const std::vector<double>& position,
+                    const std::vector<double>& velocity);
+
+private:
+    class Impl;
+    std::unique_ptr<Impl> pImpl;
+};
+
+struct scenario::base::PID
+{
+    PID() = default;
+    PID(const double _p, const double _i, const double _d)
+        : p(_p)
+        , i(_i)
+        , d(_d)
+    {}
+
+    double p = 0;
+    double i = 0;
+    double d = 0;
```

Comment:
Eventually it would make sense to already support all the options provided in ignition PIDS https://ignitionrobotics.org/api/math/6.4/classignition_1_1math_1_1PID.html ?

Reply:
Thanks for catching this! It was already in mind but the it got lost in the middle of other features. Definitely this would be a nice addition.

Reply:
Addressed in https://github.com/robotology/gym-ignition/pull/158/commits/d08c6898ea7556a042a9b88aaa7f642f58dc11e6